File: src/csharp/generator/HFCsGenerator/Types/Enumeration.cs
using System;
using System.Collections.Generic;

namespace HighFlyers.Protocol.Generator.Types
{
    class Enumeration : ObjectType
    {
        public Enumeration(string name, string[][] input)
            : base(name, input)
        {
        }

        protected override string GenerateHeader()
        {
            return "enum " + Name;
        }

        protected override IEnumerable<string> GenerateBody()
        {
            foreach (var words in Input)
            {
                if (words.Length != 1)
                    throw new Exception("Expected one word in line!");

                yield return words[0];
            }
        }
    }
}

File: src/cpp-devel-test/highflyers-protocol-generator/Enumeration.cpp
#include "Enumeration.h"

using namespace HighFlyers::Protocol;
using namespace std;

Enumeration::Enumeration(const string& name, const vector<vector<string>>& input)
	: ObjectType(name, input)
{}

string Enumeration::generate_header()
{
	return "";
}

vector<string> Enumeration::generate_body()
{
	vector<string> out;

	for (auto words : input)
	{
		if (words.size() != 1)
			throw std::runtime_error("Expected one word in line!");

		out.push_back(words[0] + ",");
	}

	return out;
}
